mappers/modbus/driver/client.go

```diff
@@ -0,0 +1,149 @@
+/*
+Copyright 2020 The KubeEdge Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package driver
+
+import (
+	"errors"
+	"time"
+
+	//. "github.com/kubeedge/kubeedge/mappers/modbus/globals"
+	"github.com/sailorvii/modbus"
+	"k8s.io/klog"
+)
+
+// ModbusTcp is the configurations of modbus TCP.
+type ModbusTcp struct {
+	SlaveId  byte
+	DeviceIp string
+	TcpPort  string
+	Timeout  time.Duration
+}
+
+// ModbusRtu is the configurations of modbus RTU.
+type ModbusRtu struct {
+	SlaveId      byte
+	SerialName   string
+	BaudRate     int
+	DataBits     int
+	StopBits     int
+	Parity       string
+	RS485Enabled bool
+	Timeout      time.Duration
+}
+
+// ModbusClient is the structure for modbus client.
+type ModbusClient struct {
+	Client  modbus.Client
+	Handler interface{}
+	Config  interface{}
+}
+
+// NewClient allocate and return a modbus client.
+// Client type includes TCP and RTU.
+func NewClient(config interface{}) (ModbusClient, error) {
+	switch config.(type) {
+	case ModbusTcp:
+		c, _ := config.(ModbusTcp)
+		handler := modbus.NewTCPClientHandler(c.DeviceIp + ":" + c.TcpPort)
+		handler.Timeout = c.Timeout
+		handler.IdleTimeout = c.Timeout
+		handler.SlaveId = c.SlaveId
+		return ModbusClient{Client: modbus.NewClient(handler), Handler: handler, Config: c}, nil
+	case ModbusRtu:
+		c, _ := config.(ModbusRtu)
+		handler := modbus.NewRTUClientHandler(c.SerialName)
+		handler.BaudRate = c.BaudRate
+		handler.DataBits = c.DataBits
+		handler.Parity = parity(c.Parity)
+		handler.StopBits = c.StopBits
+		handler.SlaveId = c.SlaveId
+		handler.Timeout = c.Timeout
+		handler.IdleTimeout = c.Timeout
+		handler.RS485.Enabled = c.RS485Enabled
+		return ModbusClient{Client: modbus.NewClient(handler), Handler: handler, Config: c}, nil
+	default:
+		return ModbusClient{}, errors.New("Wrong modbus type")
+	}
+}
+
+// GetStatus get device status.
+// Now we could only get the connection status.
+func (c *ModbusClient) GetStatus() string {
+	err := c.Client.Connect()
```

Comment:
From edgex example(also I have ask my colleague) we should close modbus connection after each run.

For example, we have two device DeviceA and DeviceB, they have same serial port(such as /dev/ttyS0) but different SlaveID.
In your code, you will create two modus client for them. And you have not closed the collection after read/write action. So if DeviceA already open /dev/ttyS0, deviceB will connect faild and have no change to success.

Reply:
Hmm, it's a complex question. The second open will succeed in real. But as you said, two slaves in one device could not visit at the same time. I'll think it more.